### PR TITLE
feat(reason_ls): add reason_ls for ReasonML

### PR DESCRIPTION
**Pull request by ri7nz**
Since This PR in `nvim-lspconfig` https://github.com/neovim/nvim-lspconfig/commit/e67f23e93214340fabf8a48fcdf46ddfec59093b merged, I think it will be better when have installer.

### CODE REVIEW

File: lua/nvim-lsp-installer/servers/reason_ls/init.lua [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+            context.use_github_release_file("jaredly/reason-language-server", archive_name..".zip"),
+            context.capture(function(ctx)
+                return std.unzip_remote(ctx.github_release_file)
+            end),
+            context.receipt(function(receipt, ctx)
+                receipt:with_primary_source(receipt.github_release_file(ctx))
+            end),
+        },
+        default_options = {
+            cmd = { path.concat { root_dir, archive_name, "reason-language-server" }},
```
Comment (williamboman) [inline lua/nvim-lsp-installer/servers/reason_ls/init.lua:31]: I see lspconfig already provide the cmd for this server. In that case it's preferred to instead provide the PATH to the executable instead, see for example clangd: https://github.com/williamboman/nvim-lsp-installer/blob/main/lua/nvim-lsp-installer/servers/clangd/init.lua#L34-L38

**Review by williamboman — COMMENTED** [review]
Awesome! Just some suggestions

File: lua/nvim-lsp-installer/servers/reason_ls/init.lua (hunk 1 above)
Comment (ri7nz) [reply lua/nvim-lsp-installer/servers/reason_ls/init.lua:31]: @williamboman  So, I must be define `cmd_env`  like `clangd` ?

Comment (williamboman) [reply lua/nvim-lsp-installer/servers/reason_ls/init.lua:31]: Exactly!

Comment (ri7nz) [reply lua/nvim-lsp-installer/servers/reason_ls/init.lua]: @williamboman  could you review new changes ?

File: lua/nvim-lsp-installer/servers/reason_ls/init.lua [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        languages = { "reason" },
+        homepage = "https://github.com/jaredly/reason-language-server",
+        installer = {
+            context.use_github_release_file("jaredly/reason-language-server", archive_name),
+            context.capture(function(ctx)
+                return std.unzip_remote(ctx.github_release_file)
+            end),
+            context.capture(function()
+              return std.rename(archive_name:gsub("%.zip", "") , "reason")
+            end),
```
Comment (williamboman) [inline lua/nvim-lsp-installer/servers/reason_ls/init.lua]: Ah my apologies - I did not see this part of the code in my earlier review. I'll revert back to what you had done before, my bad!

**Review by williamboman — APPROVED** [review]
LGTM! Thanks!

File: lua/nvim-lsp-installer/servers/reason_ls/init.lua (hunk 2 above)
Comment (ri7nz) [reply lua/nvim-lsp-installer/servers/reason_ls/init.lua]: No problem @williamboman .

**Pull request merged**